tmt/steps/report/reportportal.py

```diff
@@ -38,113 +25,179 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
         option="--project",
-        metavar="PROJECT",
+        metavar="PROJECT_NAME",
         default=None,
         help="The project name which is used to create the full URL.")
-    launch_name: Optional[str] = tmt.utils.field(
-        option="--launch-name",
-        metavar="NAME",
+    launch: Optional[str] = tmt.utils.field(
+        option="--launch",
+        metavar="LAUNCH_NAME",
+        default=None,
+        help="The launch name (name of plan per launch is used by default).")
+    env_var: Optional[str] = tmt.utils.field(
+        option="--env-var",
+        metavar="TMT_VAR",
         default=None,
-        help="The launch name (base name of run id used by default).")
+        multiple=True,
+        help="List one of the environment variables that begin with \"TMT_\" to disobey \
+the rule of not displaying them in Report Portal launch (the argument can be repeated).")
 
 
 @tmt.steps.provides_method("reportportal")
 class ReportReportPortal(tmt.steps.report.ReportPlugin):
     """
-    Report test results to a ReportPortal instance
+    Report test results to a ReportPortal instance via reportportal_client API
 
     Requires a TOKEN for authentication, a URL of the ReportPortal
-    instance and the PROJECT name. In addition to command line options
-    it's possible to use environment variables to set the url and token:
+     instance and the PROJECT name. In addition to command line options
+     it's possible to use environment variables to set the url and token:
 
         export TMT_REPORT_REPORTPORTAL_URL=...
         export TMT_REPORT_REPORTPORTAL_TOKEN=...
 
-    The optional launch NAME is passed to ReportPortal. Assuming the URL
-    and TOKEN variables are provided by the environment, the config can
-    look like this:
+    The optional launch NAME doesn't have to be provided if it is the same
+     as the plan name (by default). Assuming the URL and TOKEN variables
+     are provided by the environment, the config can
+     look like this:
 
         report:
             how: reportportal
             project: baseosqe
-            launch-name: maven
+
+    In item details of the ReportPortal instance, the environment variables
+     per test are displayed except those that begin with ``TMT_``
+     (unique variables break history aggregation). To obey this rule, list
+     these variables for particular use case with option --env-vars
+            env-vars: [TMT_TEST_DATA,TMT_PLAN_DATA]
     """
 
     _data_class = ReportReportPortalData
 
+    def handle_response(self, response: requests.Response) -> None:
+        if not response.ok:
+            raise tmt.utils.ReportError(f"\n\
+Received non-ok status code from ReportPortal. Response text:\n{response.text}")
+        else:
+            self.debug(f"Response code from the server: {response.status_code}")
+            self.debug(f"Message from the server: {response.text}")
```

Comment:
You can use the native key/value approach `debug()` supports: `debug('Response code from the server', str(response.status_code))`.

Reply:
Also, `else` is not needed: should the condition in `if`  be successfull, an exception wuld be raised and we would leave this function. THe following would be equivalent with one less level of indentation, which is always better for readability:

```python
if not response.ok:
    raise tmt.utils.ReportError(f"\nReceived non-ok status code from ReportPortal. Response text:\n{response.text}")

self.debug(f"Response code from the server: {response.status_code}")
self.debug(f"Message from the server: {response.text}")
```